h2o-algos/src/main/java/hex/gam/GAMModel.java

```diff
@@ -536,32 +545,90 @@ public static Frame addTPGamColumns(Frame adaptedF, GAMParameters parms, double[
     return concateGamVecs(addTPCols._gamFrameKeysCenter);
   }
   
-  public static Frame addCSGamColumns(Frame adptedF, GAMParameters parms, String[][] gamColNames,
-                                      double[][][] binvD, double[][][] zTranspose, double[][][] knots) {
+  public static Frame addSingleVariableGamColumns(Frame adptedF, GAMParameters parms, String[][] gamColNames,
+                                                  double[][][] binvD, double[][][] zTranspose, double[][][] knots, 
+                                                  int numCSGamCol) {
     int numGamCols = parms._gam_columns.length;
-    int numCSGamCols = numGamCols - (parms._M==null?0:parms._M.length);
-    if (numCSGamCols == 0)
+    int numISplineGamCol = numGamCols - numCSGamCol-(parms._M==null?0:parms._M.length);
+    int numSingleVariableGamCols = numGamCols - (parms._M==null?0:parms._M.length);
```

Comment:
could be `int numSingleVariableGamCols = numISplineGamCol  + numCSGamCol`

Reply:
You are right.  Changed.